### PR TITLE
refactor: review notes + webhooks PLATFORM-8899 

### PR DESCRIPTION
**Pull request by mmeller-wikia**
See https://github.com/Wikia/kratos/pull/95 and https://github.com/Wikia/kratos/pull/96

### CODE REVIEW

File: identity/handler.go [hunk 1]
```diff
@@ -734,6 +736,72 @@ func (h *Handler) validate(w http.ResponseWriter, r *http.Request, _ httprouter.
...
+//	Schemes: http, https
+//
+//	Security:
+//	  oryAccessToken:
+//
+//	Responses:
+//	  204: emptyResponse
+//	  404: jsonError
+//	  500: jsonError
+func (h *Handler) deleteMultifactorCredential(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
```
Comment (mmeller-wikia) [inline identity/handler.go:777]: Note: It should be removed once we migrate to

File: selfservice/hook/web_hook.go [hunk 2]
```diff
@@ -204,6 +207,7 @@ func (e *WebHook) ExecutePostRecoveryHook(_ http.ResponseWriter, req *http.Reque
 			RequestURL:     x.RequestURL(req).String(),
 			RequestCookies: cookies(req),
 			Identity:       session.Identity,
+			HookType:       "RecoveryPostHook",
```
Comment (mmeller-wikia) [inline selfservice/hook/web_hook.go:210]: Note: Previously it was `PostRecoveryHook`

File: selfservice/hook/web_hook.go [hunk 3]
```diff
@@ -354,6 +367,8 @@ func (e *WebHook) ExecuteSettingsPostPersistHook(_ http.ResponseWriter, req *htt
 			RequestURL:     x.RequestURL(req).String(),
 			RequestCookies: cookies(req),
 			Identity:       id,
+			Credentials:    credentials,
+			HookType:       "SettingsPostPersistHook",
```
Comment (mmeller-wikia) [inline selfservice/hook/web_hook.go:371]: Issue: 
```suggestion
			HookType:       "SettingsPostPersistHook:" + settingsType,
```

File: driver/registry_default_settings.go [hunk 4]
```diff
@@ -11,7 +11,7 @@ import (
 )
 
 func (m *RegistryDefault) PostSettingsPrePersistHooks(ctx context.Context, settingsType string) (b []settings.PostHookPrePersistExecutor) {
-	for _, v := range m.getHooks(settingsType, m.Config().SelfServiceFlowSettingsAfterHooks(ctx, settingsType)) {
+	for _, v := range m.getHooks(settingsType, filter(m.Config().SelfServiceFlowSettingsAfterHooks(ctx, settingsType), config.PrePersist)) {
```
Comment (mmeller-wikia) [inline driver/registry_default_settings.go:14]: TODO: check if all `filter` calls are in place

File: selfservice/hook/web_hook.go [hunk 5]
```diff
@@ -484,7 +509,8 @@ func (e *WebHook) execute(ctx context.Context, data *templateContext) error {
 		if resp.StatusCode >= http.StatusBadRequest {
 			span.SetStatus(codes.Error, "HTTP status code >= 400")
 			if canInterrupt || parseResponse {
-				if err := parseWebhookResponse(resp, data.Identity); err != nil {
+				// TODO: double-check if we could use upstream `parseWebhookResponse`
```
Comment (mmeller-wikia) [inline selfservice/hook/web_hook.go:512]: TODO:

File: selfservice/hook/web_hook.go [hunk 6]
```diff
@@ -389,9 +404,19 @@ func (e *WebHook) ExecuteSettingsPrePersistHook(_ http.ResponseWriter, req *http
...
 	var (
-		httpClient     = e.deps.HTTPClient(ctx)
+		httpClient = e.deps.NamedHTTPClient(
+			ctx,
+			data.HookType+conf.sum,
+			httpx.ResilientClientWithMaxRetry(conf.retries),
+			httpx.ResilientClientWithConnectionTimeout(conf.timeout),
+			httpx.ResilientClientWithMinxRetryWait(conf.minWait),
+			httpx.ResilientClientWithMaxRetryWait(conf.maxWait),
+		)
```
Comment (mmeller-wikia) [inline selfservice/hook/web_hook.go:419]: Maybe we could push this change to upstream

Comment (abador) [reply selfservice/hook/web_hook.go:419]: I think we should add ACs to all such tickets

**Review by adpaste — APPROVED** [review]

**Pull request merged**